### PR TITLE
Add libvault 0.48.0

### PR DESCRIPTION
**Pull request by sonicxml**
Specify library name and version:  **libvault/0.48.0**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

Not the author, but I wish to use this library in my existing Conan project. 

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by sonicxml** [comment]
@uilianries could I please get some help here if you get a chance? It looks like CI is trying to build with an older version of clang (getting `fatal error: 'filesystem' file not found #include <filesystem>`) but I thought I specified only using a newer clang that includes the `filesystem` header

**Comment by SSE4** [comment]
> @uilianries @SSE4 could I please get some help here if you get a chance? It looks like CI is using a current version of clang but I am still getting `fatal error: 'filesystem' file not found #include <filesystem>`
> 
> Compiling with filesystem and clang 11 on godbolt works though: https://godbolt.org/z/v9b389M5a

see comments: https://github.com/conan-io/conan-center-index/pull/9063#issuecomment-1020446658 (and below)
TLDR: you can exclude clang + libstdc++ for now

File: recipes/libvault/all/patches/fix-cmake-0.48.0.patch [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+ if (ENABLE_TEST)
+-  include(FetchContent)
+-  FetchContent_Declare(
+-          Catch2
+-          GIT_SHALLOW    TRUE
+-          GIT_REPOSITORY https://github.com/catchorg/Catch2.git
+-          GIT_TAG        v2.13.6)
+-  FetchContent_MakeAvailable(Catch2)
+-  list(APPEND CMAKE_MODULE_PATH ${Catch2_SOURCE_DIR}/contrib)
+   enable_testing()
```
Comment (uilianries) [inline recipes/libvault/all/patches/fix-cmake-0.48.0.patch:69]:
```suggestion
           COMPATIBILITY SameMajorVersion)
-
-  install(FILES "${CMAKE_CURRENT_BINARY_DIR}/libvaultConfig.cmake"
-          "${CMAKE_CURRENT_BINARY_DIR}/libvaultConfigVersion.cmake"
-          DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/libvault")
-
-  configure_file(vault.pc.in vault.pc @ONLY)
-  install(FILES "${CMAKE_BINARY_DIR}/${PROJECT_NAME}.pc"
-          DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig")
 endif (INSTALL)

 if (ENABLE_TEST)
-  include(FetchContent)
-  FetchContent_Declare(
-          Catch2
-          GIT_SHALLOW    TRUE
-          GIT_REPOSITORY https://github.com/catchorg/Catch2.git
-          GIT_TAG        v2.13.6)
-  FetchContent_MakeAvailable(Catch2)
-  list(APPEND CMAKE_MODULE_PATH ${Catch2_SOURCE_DIR}/contrib)
   enable_testing()
```

Don't need to patch this part, you can use tools.rmdir as other recipes and remove the produced cmake folder.

File: recipes/libvault/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+
+    def package(self):
+        self.copy("LICENSE", dst="licenses", src=self._source_subfolder)
+        cmake = self._configure_cmake()
+        cmake.install()
+
+    def package_info(self):
+        self.cpp_info.libs = tools.collect_libs(self)
+        self.cpp_info.filenames["cmake_find_package"] = "libvault"
+        self.cpp_info.filenames["cmake_find_package_multi"] = "libvault"
```
Comment (uilianries) [inline recipes/libvault/all/conanfile.py:109]:
```suggestion
        self.cpp_info.names["cmake_find_package"] = "libvault"
        self.cpp_info.names["cmake_find_package_multi"] = "libvault"
        self.cpp_info.names["pkg_config"] = "vault"
```

File: recipes/libvault/all/patches/fix-cmake-0.48.0.patch (hunk 1 above)
Comment (sonicxml) [reply recipes/libvault/all/patches/fix-cmake-0.48.0.patch]: changed

File: recipes/libvault/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+            tools.patch(**patch)
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        self.copy("LICENSE", dst="licenses", src=self._source_subfolder)
+        cmake = self._configure_cmake()
+        cmake.install()
+
+        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
```
Comment (uilianries) [inline recipes/libvault/all/conanfile.py:106]:
```suggestion
        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
```

**Review by uilianries — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]
